### DISCUSSION TITLE
AMD APU support

### DISCUSSION THREAD

**Discussion by pauljurczak** [AdaptiveCpp/AdaptiveCpp#1301] [General]
The readme file lists only *AMD GPUs that are supported by ROCm* as supported hardware. I think that clang supports intrinsics for many AMD APUs. Also I found this *System 1: AMD Ryzen 7 4750U APU, 32GB RAM, Arch Linux* on page 17 of https://www.iwocl.org/wp-content/uploads/iwocl-2023-Aksel-Alpay-2566.pdf.

What is the current and near future level of support for the newest AMD APUs with RDNA 3?

**Comment by illuhad** [comment]
All of these statements are true at the same time :-)

- AdaptiveCpp sits on top of ROCm, so generally, the AMD hardware that works in AdaptiveCpp is the AMD hardware that works in ROCm.
- AMD refuses to officially support more than a handful of GPUs in ROCm. GPUs not on the ROCm support list are not explicitly validated by AMD, but in practice ROCm still mostly works on them (but sometimes doesn't).

So, GPUs that are not officially supported in ROCm have a good chance of working with AdaptiveCpp. However, we cannot guarantee it because if there's an issue at the ROCm or driver level, we cannot fix it as those issues usually can only be fixed inside ROCm.

In practice, all AMD GPUs are usually worth a try.

>  I think that clang supports intrinsics for many AMD APUs

LLVM is not the only component that needs to work. Indeed, LLVM generally supports AMD hardware across the board. But there's also bitcode libraries that come with ROCm that implement math or special functions, there's the HIP/HSA runtime library and there's the driver.
